src/app/components/audio-recordings/pages/analysis-results/analyses-results.component.ts

```diff
@@ -0,0 +1,193 @@
+   */
+  public getItem(model?: ResultNode): Observable<AnalysisJobItemResult> {
+    const analysisJobId = this.analysisJob;
+    return this.api.show(model.result, analysisJobId, this.audioRecording.id);
+  }
+
+  private closeRow(model: ResultNode): void {
+    model.children?.forEach(child => {
+      // remove the child element from the rows
+      this.rows.splice(this.rows.indexOf(child), 1);
+
+      // close all the child elements of the row
+      this.closeRow(child)
+    });
+
+    model.open = false;
+  }
+
+  private openRow(model: ResultNode): void {
+    this.getModelChildren(model)
+      // since we have evaluated the children of the model, append this information to the model
+      // so that we don't have to re-query this information in the future from the API
+      .pipe(map(children => model.children = children))
+      .pipe(map(returnedValues => this.rows.splice(this.rows.indexOf(model) + 1, 0, ...returnedValues)))
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe();
+
+    model.open = true;
+  }
+
+  public toggleRow(model: ResultNode): void {
+    if (model.open) {
+      this.closeRow(model);
+    } else {
+      this.openRow(model);
+    }
+  }
+
+  /**
+   * returns a models child items by evaluating the object using the baw-api and adds path information to the model
+   * this helper method is intended to take a partial model, as is present in a complete models children attribute.
+   *
+   * @param model A model with a name attribute to evaluate the child items of
+   * @returns An array of type `ResultNode` representing the child items, of the model, the child items parent, and their path
+   */
+  public getModelChildren(
+    model: ResultNode,
+  ): Observable<ResultNode[]> {
+    return new Observable<ResultNode[]>(
+      subscriber => {
+        this.getItem(model)
+          // add the path & parent information to all child items
+          .pipe(map(returnedValue => this.childItemsWithParentInformation(returnedValue)))
+          .pipe(takeUntil(this.unsubscribe))
+          .subscribe({
+            next(modelChildren) {
+              subscriber.next(modelChildren);
+            }
+          });
+      }
+    );
+  }
+
+  /**
+   * Takes a view model and returns the child items, with the `path`, `analysisJobId`, and `audioRecordingId` attributes
+   *
+   * @param model A view model to fetch the children of
+   * @returns An array of `AnalysisJobItemResult` representing the child items inside the model
+   */
+  private childItemsWithParentInformation(model: AnalysisJobItemResult): ResultNode[] {
+    const evaluatedSubItems = Array<ResultNode>();
+
+    model.children.forEach(item =>
+      evaluatedSubItems.push({
+        parentItem: model,
+        result: new AnalysisJobItemResult({
+          type: "file",
+          path: model.path + item.name,
+          ...item
+        }),
```

Comment:
The AnalysisJobItemResult model doesn't have the information on the parent node / model, thus why I placed this on the `ResultNode` interface. The path / parent information also needs to be communicated to the `analysis-download-row.component.ts`, which is why I added it to the result model.

The path information can be derived using the parent node + item name, but this information will need to be communicated to the `analysis-download-row.component.ts`, which will require the `ResultNode` interface to be replicated on the `analysis-download-row.component.ts`.

Reply:
so; 

i'm not sure what information is missing; show me in a call

but you didn't answer why you're recreating the model

another design: a function on the model `getFullPath(parent)`

Reply:
The need for the path information to be transferred between `analyses-results.component.ts` and `analysis-download-row.component.ts` has been removed since indentation can be included in the `ResultNode` interface. Removing the need poor code quality that lead to this situation.